### DISCUSSION TITLE
Should vcs-index.json be committed or ignored?

### DISCUSSION THREAD

**Discussion by HughWarrington** [joyfullservice/msaccess-vcs-addin#541] [Q&A] [answered]
vcs-index.json contains various fields which should be local only to a particular checkout of the repo and are liable to churn over time, like `FullBuildDate`, and therefore I assume this file should not be committed to version control, and I should add it to my .gitignore.

On the other hand I'm not sure whether this file is crucial to making sure export/import works at all, or whether it's just an optimisation to avoid duplicate work by allowing incremental diffs instead of starting from scratch every time.

Any ideas? It would be good to mention this on the wiki. By far the best documentation I have found so far on how to use this plugin in conjuction with git is https://github.com/joyfullservice/msaccess-vcs-addin/discussions/405, perhaps that could be folded into a wiki page. I'd be happy to write it if I had edit privileges.

**Comment by joyfullservice** [comment, marked as answer]
Yes, you are correct that `vcs-index.json` is a local file used to index and compare changes to source and database objects. It is automatically created during export, and improves the performance of export/merge operations by avoiding unnecessary processing of unchanged objects. This file is included in the default `.gitignore` and should be excluded from version control.

The `Wiki` folder in this repository is (manually) mirrored over to the actual wiki on the project, so feel free to create a pull request with any suggested changes to the files in the Wiki folder. Thanks for helping make this project better for everyone!  😄

**Reply by HughWarrington** [reply]
Great, thanks for the answer.

I think I already had my own .gitignore in my repo, which is maybe why my first export didn't create the default one.

I will make some updates to the docs when I feel like the dust has settled and I better know what I'm doing :)
